it/EditSetRecordISpec.scala

```diff
@@ -0,0 +1,3489 @@
+            response,
+            OK,
+            generateExpectedEditRecordPageFromRecord(editSetRecordOci)
+              .copy(
+                coveringDates = "1868; 1890-1902; 1933",
+                startDate = ExpectedDate("1", "1", "1868"),
+                endDate = ExpectedDate("31", "12", "1933")
+              )
+          )
+
+          assertPageAsExpected(
+            getEditSetRecordEditPageWhileLoggedIn(idOfExistingEditSet, editSetRecordOci),
+            OK,
+            generateExpectedEditRecordPageFromRecord(editSetRecordOci)
+              .copy(coveringDates = "1962")
+          )
+
+        }
+      }
+    }
+  }
+
+  private def getEditSetRecordEditPageWhileLoggedOut(
+    editSetId: String = idOfExistingEditSet,
+    editSetRecordOci: String = ociOfExistingRecord
+  ): WSResponse =
+    getEditSetRecordEditPage(editSetId, editSetRecordOci, None)
+
+  private def getEditSetRecordEditPageWhileLoggedIn(
+    editSetId: String = idOfExistingEditSet,
+    editSetRecordOci: String = ociOfExistingRecord
+  ): WSResponse =
+    getEditSetRecordEditPage(editSetId, editSetRecordOci, Some(loginForSessionCookie()))
+
+  private def getEditSetRecordEditPage(
+    editSetId: String,
+    editSetRecordOci: String,
+    sessionCookie: Option[WSCookie]
+  ): WSResponse = {
+    val rawUrl =
+      s"/edit-set/$editSetId/record/$editSetRecordOci/edit"
+    val baseUrl = wsUrl(rawUrl).withFollowRedirects(false)
+    val url = sessionCookie.map(baseUrl.addCookies(_)).getOrElse(baseUrl)
+    await(url.get())
+  }
+
+  private def getEditSetRecordDiscardPage(
+    editSetId: String,
+    editSetRecordOci: String
+  ): WSResponse = {
+    val rawUrl =
+      s"/edit-set/$editSetId/record/$editSetRecordOci/edit/discard"
+    val baseUrl = wsUrl(rawUrl).withFollowRedirects(false)
+    val url = baseUrl.addCookies(loginForSessionCookie())
+    await(url.get())
+  }
+
+  private def getEditSetRecordSavePage(
+    editSetId: String,
+    editSetRecordOci: String
+  ): WSResponse = {
+    val rawUrl =
+      s"/edit-set/$editSetId/record/$editSetRecordOci/edit/save"
+    val baseUrl = wsUrl(rawUrl).withFollowRedirects(false)
+    val url = baseUrl.addCookies(loginForSessionCookie())
+    await(url.get())
+  }
+
+  private def getExpectedEditSetRecord(oci: String): EditSetRecord =
+    editSetRecordMap.getOrElse(oci, fail(s"Unable to get record for OCI [$oci]"))
+
+  private def generateExpectedEditRecordPageFromRecord(oci: String): ExpectedEditRecordEditPage = {
+    val editSetRecord = getExpectedEditSetRecord(oci)
+    val messages: Map[String, String] = Map("edit-set.record.edit.type.physical" -> "Physical Record")
+    val expectedEditRecordPage = ExpectedEditRecordEditPage(
+      title = "Edit record",
+      heading = s"TNA reference: ${editSetRecord.ccr}",
+      subHeading = s"PAC-ID: ${editSetRecord.oci} ${editSetRecord.recordType match {
+          case Some(PhysicalRecord) => messages.get("edit-set.record.edit.type.physical").get
```

Comment:
I think you can do
```
messages("edit-set.record.edit.type.physical")
```

Reply:
That was copied as in from another Spec, but I can certainly change it here.

Reply:
Changed.